src/components/tx/SignOrExecuteForm/index.tsx

```diff
@@ -155,16 +158,15 @@ const SignOrExecuteForm = ({
   }
 
   // Execute transaction
-  const onExecute = async (): Promise<string> => {
+  const onExecute = async (): Promise<string | undefined> => {
     const [, createdTx, provider] = assertDependencies()
 
     // If no txId was provided, it's an immediate execution of a new tx
-    const id = txId || (await proposeTx(createdTx))
```

Comment:
I believe we also still need to propose when executing with a smart contract wallet e.g. a 1/n safe has another safe as owner

Reply:
I think this is fine as it's regarding execution. Final confirmations/immediate executions eventually wind up on chain where they will be indexed by our services.

Reply:
Yes, that still works but its now missing notifications on the safe so it never displays that the transaction was successfully executed. I don't see a reason to change this behaviour for execution but not for signing with a smart contract wallet.

Reply:
The reason is described in the issue. On-chain signatures just happen not to work at all when creating a new transaction. An alternative that I'm considering would be to disable tx creation for SC wallets.

Reply:
The difference in behaviour I mean is that when I sign the transaction in Safe A and execute the confirmation immediately in Safe B, it shows notifications in Safe A saying "An on-chain signature is required. Please confirm the transaction in your wallet.". This one is missing when immediately executing the transaction in Safe A even though it also has to wait for the other transaction to be executed in Safe B.

Another case is that if both transactions are immediately executes, previously it would show "Your transaction was successfully executed." in Safe A but now it doesn't which seems like a regression to me.

Reply:
"Your transaction was successfully executed." isn't shown now because it comes from the backend. Once a tx lands in the history, we show this notification, but only if there's a pending tx with this id. Without proposing, there's no id.
So it's not a regression, it's a logical consequence of not proposing on creation+immediate execution. And it's the same for SC wallets and regular wallets.

Reply:
My argument ist that whether it is an immediate execution or just signing, Safe A will always have to wait for another transaction to be executed in Safe B. This could happen immediately (if Safe B also immediately executes) or could take days/weeks or longer. Why do we propose in one of the cases but not the other?

Reply:
We propose only for SC wallets because if we don't, the backend cannot match the tx and the sig from approveHash.

Reply:
Ok, that makes sense. So in case of an immediate execution, the transaction in Safe B contains the execution of Safe A so its all in one place and doesn't need a reference like the sign tx does.